Content.Client/Storage/Visualizers/MappedItemVisualizer.cs

```diff
@@ -0,0 +1,107 @@
+
+                foreach (var (sprite, _) in _spriteLayers)
+                {
+                    spriteComponent.LayerMapReserveBlank(sprite);
+                    spriteComponent.LayerSetSprite(sprite, new SpriteSpecifier.Rsi(_rsiPath, sprite));
+                    spriteComponent.LayerSetVisible(sprite, false);
+                }
+            }
+        }
+
+        public void AfterDeserialization()
+        {
+            if (_mapLayers is { Count: > 0 })
+            {
+                foreach (var layerProp in _mapLayers)
+                {
+                    if (_spriteLayers.ContainsKey(layerProp.Layer))
+                    {
+                        Logger.Warning($"Already added mapLayer with layer = `${layerProp.Layer}` skipping over");
+                    }
+                    else
+                    {
+                        _spriteLayers.Add(layerProp.Layer, layerProp);
+                    }
+                }
+            }
+        }
+
+        public override void OnChangeData(AppearanceComponent component)
+        {
+            base.OnChangeData(component);
+            if (component.Owner.TryGetComponent<ISpriteComponent>(out var spriteComponent)
+                && _spriteLayers.Count > 0)
+            {
+                UpdateAllContainedEntities(component, spriteComponent);
+                if (component.TryGetData(StorageMapVisuals.LayerChanged, out ShowEntityData layerData))
+                {
+                    foreach (var (layerName, layerFilter) in _spriteLayers)
+                    {
+                        if (component.Owner.EntityManager.TryGetEntity(layerData.Uid, out var entity)
+                            && Matches(entity, layerFilter))
+                        {
+                            spriteComponent.LayerSetVisible(layerName, layerData.Show);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void UpdateAllContainedEntities(AppearanceComponent component, ISpriteComponent spriteComponent)
+        {
+            if (component.TryGetData(StorageMapVisuals.AllLayers, out ListOfUids listOfUids)
+                && listOfUids.ContainedEntities.Count > 0)
+            {
+                foreach (var entityUid in listOfUids.ContainedEntities)
+                {
+                    foreach (var (layerName, layerFilter) in _spriteLayers)
+                    {
+                        if (component.Owner.EntityManager.TryGetEntity(entityUid, out var entity)
+                            && Matches(entity, layerFilter))
+                        {
+                            spriteComponent.LayerSetVisible(layerName, true);
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool Matches(IEntity entity, SharedMapLayerData layerFilter)
+        {
+            if (entity.Prototype != null && entity.Prototype.ID.Equals(layerFilter.Id))
```

Comment:
Don't rely on prototype IDs please. If I have a RedCrayon or MultitoolGreen and inherit suddenly none of this functionality works when it should.